operations/ftdc.go

```diff
@@ -0,0 +1,286 @@
+				defer jsonFile.Close()
+			}
+
+			var iter ftdc.Iterator
+			if c.Bool("flattened") {
+				iter = ftdc.ReadMetrics(ctx, ftdcFile)
+			} else {
+				iter = ftdc.ReadStructuredMetrics(ctx, ftdcFile)
+			}
+
+			for iter.Next(ctx) {
+				jsonFile.WriteString(iter.Document().ToExtJSON(false))
+				jsonFile.WriteString("\n")
+			}
+
+			return iter.Err()
+		},
+	}
+}
+
+func jsontoftdc() cli.Command {
+	return cli.Command{
+		Name:  "jsontoftdc",
+		Usage: "write FTDC data from a JSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write JSON data from this file, defaults to stdin",
+			},
+			cli.StringFlag{
+				Name:  "prefix",
+				Usage: "prefix for FTDC filenames",
+			},
+			cli.IntFlag{
+				Name:  "maxChunkSize",
+				Usage: "maximum chunk size",
+				Value: 1000,
+			},
+			cli.StringFlag{
+				Name:  "flushInterval",
+				Usage: "flush interval in ms",
+				Value: "20",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			opts := ftdc.CollectJSONOptions{}
+
+			jsonPath := c.String("input")
+			if jsonPath == "" {
+				opts.InputSource = os.Stdin
+			} else {
+				opts.FileName = jsonPath
+			}
+
+			ftdcPrefix := c.String("prefix")
+			if ftdcPrefix == "" {
+				return errors.New("prefix is not specified")
+			} else {
+				opts.OutputFilePrefix = ftdcPrefix
+			}
+
+			flushInterval, err := time.ParseDuration(c.String("flushInterval") + "ms")
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse duration '%s'", c.String("flushInterval"))
+			} else {
+				opts.FlushInterval = flushInterval
+			}
+
+			opts.ChunkSizeBytes = c.Int("maxChunkSize")
+
+			if err := ftdc.CollectJSONStream(ctx, opts); err != nil {
+				return errors.Wrap(err, "Failed to write FTDC from JSON")
+			}
+			return nil
+		},
+	}
+}
+
+func ftdctobson() cli.Command {
+	return cli.Command{
+		Name:  "ftdctobson",
+		Usage: "write FTDC data to a BSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write FTDC data from this file",
+			},
+			cli.StringFlag{
+				Name:  "output",
+				Usage: "write FTDC data in BSON format to this file; defaults to stdout",
+			},
+			cli.BoolFlag{
+				Name:  "flattened",
+				Usage: "flatten FTDC data",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			ftdcPath := c.String("input")
+			if ftdcPath == "" {
+				return errors.New("input is not specified")
+			}
+			bsonPath := c.String("output")
+
+			ftdcFile, err := os.Open(ftdcPath)
+			if err != nil {
+				return errors.Wrapf(err, "problem opening file '%s'", ftdcPath)
+			}
+			defer ftdcFile.Close()
+
+			var bsonFile *os.File
+			if bsonPath == "" {
+				bsonFile = os.Stdout
+			} else {
+				bsonFile, err = os.Create(bsonPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening flie '%s'", bsonPath)
+				}
```

Comment:
you could do something like: 

defer func() { grip.Warning(errors.Wrap(bsonFile.Close())) }() 

to avoid needing to have the warning line in other places.

Reply:
done